test/harness/async_index.js

```diff
@@ -260,29 +260,40 @@ function assert_trap(action) {
     .catch(_ => {});
 }
 
-function assert_return(action, expected) {
+function assert_return(action, ...expected) {
   const test = "Test that a WebAssembly code returns a specific result";
   const loc = new Error().stack.toString().replace("Error", "");
   chain = Promise.all([action(), chain])
     .then(
       values => {
         uniqueTest(_ => {
           let actual = values[0];
-          switch (expected) {
-          case "nan:canonical":
-          case "nan:arithmetic":
-              // Note that JS can't reliably distinguish different NaN values,
-              // so there's no good way to test that it's a canonical NaN.
-              assert_true(Number.isNaN(actual), `expected NaN, observed ${actual}.`);
-              return;
-          case "ref.func":
-              assert_true(typeof actual === "function", `expected Wasm function, got ${actual}`);
-              return;
-          case "ref.any":
-              assert_true(actual !== null, `expected Wasm reference, got ${actual}`);
-              return;
-          default:
-              assert_equals(actual, expected);
+          if (actual === undefined) {
```

Comment:
I'm not sure if I merged this correctly.

Reply:
LGTM